pkg/exporter/metrics.go

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package exporter
+
+import (
+	"io"
+
+	"github.com/cilium/tetragon/pkg/metrics/consts"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	eventsExportedTotal = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: consts.MetricsNamespace,
+		Name:      "events_exported_total",
+		Help:      "Total number of events exported",
+	})
+
+	eventsExportedBytesTotal = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace: consts.MetricsNamespace,
+		Name:      "events_exported_bytes_total",
+		Help:      "Number of bytes exported for events",
+	})
+
+	eventsExportTimestamp = prometheus.NewGauge(prometheus.GaugeOpts{
+		Namespace: consts.MetricsNamespace,
+		Name:      "events_last_exported_timestamp",
+		Help:      "Timestamp of the most recent event to be exported",
+	})
+)
+
+func InitMetrics(registry *prometheus.Registry) {
+	registry.MustRegister(eventsExportedTotal)
+	registry.MustRegister(eventsExportedBytesTotal)
+	registry.MustRegister(eventsExportTimestamp)
+}
+
+func newExportedBytesCounterWriter(w io.Writer, c prometheus.Counter) io.Writer {
+	return byteCounterWriter{Writer: w, bytesWritten: c}
+}
+
+type byteCounterWriter struct {
+	io.Writer
+	bytesWritten prometheus.Counter
```

Comment:
do we need to pass the metrics like that? can't we just increment the eventsExportedBytesTotal directly?

Reply:
If we wanted this to be a global singleton implementation, perhaps. However that would likely lead to mistakes later if the same type was used somewhere else.